fdbserver/BlobWorker.actor.cpp

```diff
@@ -2859,21 +2909,24 @@ ACTOR Future<Void> blobGranuleUpdateFiles(Reference<BlobWorkerData> bwData,
 			}
 			justDidRollback = false;
 
-			// Write a new delta file IF we have enough bytes OR force flush.
+			// xch: Write a new delta file IF we have enough bytes OR force flush.
 			// The force flush contract is a version cannot be put in forceFlushVersion unless the change feed
 			// is already whenAtLeast that version
 			// FIXME: with non-uniform delta file sizes, we could potentially over-shoot target write amp
 			bool forceFlush = !forceFlushVersions.empty() && forceFlushVersions.back() > metadata->pendingDeltaVersion;
 			bool doEarlyFlush = !metadata->currentDeltas.empty() && metadata->doEarlyReSnapshot();
 			CODE_PROBE(forceFlush, "Force flushing granule");
-			if ((processedAnyMutations && metadata->bufferedDeltaBytes >= writeAmpTarget.getDeltaFileBytes()) ||
-			    forceFlush || doEarlyFlush) {
+
+			CODE_PROBE(flushTopMemory, "Flush top granules using the most memory");
+
+			if ((processedAnyMutations && metadata->bufferedDeltaBytes >= writeAmpTarget.bytesBeforeCompact) ||
```

Comment:
metadata->bufferedDeltaBytes >= writeAmpTarget.bytesBeforeCompact

This place need to be discussed, my goal is to prevent a delta file that is too big